src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.tsx

```diff
@@ -1,196 +1,365 @@
+  amount: yup
+    .number()
+    .min(0, 'Amount must be positive')
+    .required('Amount is required'),
+});
+
 export const GoalCalculatorGrid: React.FC<GoalCalculatorGridProps> = ({
-  formData,
+  category,
   promptText,
-  categoryName,
 }) => {
-  const { handleContinue } = useGoalCalculator();
   const { t } = useTranslation();
+  const locale = useLocale();
+  const { label: categoryName } = category;
+  const accountListId = useAccountListId() ?? '';
 
-  const initialValues: GoalCalculatorGridFormValues = {
-    gridData: formData || [
-      { id: 1, name: 'Freelance Work', amount: 2500 },
-      { id: 2, name: 'Investment Returns', amount: 1200 },
-      { id: 3, name: 'Rental Income', amount: 1800 },
-    ],
-    lumpSumAmount: 0,
-  };
+  const gridData = React.useMemo(
+    () =>
+      category.subBudgetCategories.map((subCategory) => ({
+        id: subCategory.id,
+        label: subCategory.label,
+        amount: subCategory.amount,
+        canDelete: !subCategory.category,
+      })),
+    [category.subBudgetCategories]
+  );
+
+  const totalAmount = gridData.reduce((sum, item) => sum + item.amount, 0);
+  const [cellErrors, setCellErrors] = useState<
+    Record<string, string | undefined>
+  >({});
+  const [directInputError, setDirectInputError] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string>('');
+  const [updatePrimaryBudgetCategory] =
+    useUpdatePrimaryBudgetCategoryMutation();
+  const [localDirectInput, setLocalDirectInput] = useState<boolean | null>(
+    null
+  );
+  const [updateSubBudgetCategory] = useUpdateSubBudgetCategoryMutation();
+  const [createSubBudgetCategory] = useCreateSubBudgetCategoryMutation();
+  const [deleteSubBudgetCategory] = useDeleteSubBudgetCategoryMutation();
 
-  const validationSchema = yup.object({
-    gridData: yup
-      .array()
-      .of(
-        yup.object({
-          id: yup.number().required(),
-          name: yup
-            .string()
-            .min(2, t('Name must be at least 2 characters'))
-            .required(t('Name is required')),
-          amount: yup
-            .number()
-            .min(0, t('Amount must be positive'))
-            .required(t('Amount is required')),
-        }),
-      )
-      .optional(),
-    lumpSumAmount: yup.number().min(0, t('Amount must be positive')).optional(),
-  });
-
-  const handleSubmit = () => {
-    // Handle form submission here
-    // TODO: Implement form submission logic
-    handleContinue();
+  const dataWithTotal = [
+    ...gridData,
+    { id: 'total', label: 'Total', amount: totalAmount },
+  ];
+
+  const directInput =
+    localDirectInput !== null ? localDirectInput : !!category.directInput;
+  const lumpSumAmount = category.directInput || 0;
+
+  useEffect(() => {
+    if (directInput) {
+      setInputValue(lumpSumAmount.toString());
+    }
+  }, [lumpSumAmount]);
+
+  const updatePrimaryBudgetCategoryMutation = (value: number | null) => {
```

Comment:
What do you think about this name?

```suggestion
  const updateDirectInput = (directInput: number | null) => {
```